### ISSUE TITLE
Update Checker should be opt OUT, not opt IN. 

### ISSUE THREAD

**Issue by cculianu** [Electron-Cash/Electron-Cash#1142]
As I'm sure you are aware, the phishers are up to no good. Again!

Right now we have the update checker: 

<img width="455" alt="screen shot 2019-02-02 at 12 13 39 pm" src="https://user-images.githubusercontent.com/266627/52162991-03d1f080-26e4-11e9-984c-caab592daceb.png">

It's off by default and needs to be found buried in the settings.  This is not acceptable, IMHO. We need it ON by default.

Right now we are good because all the sybil attackers can do is phish.  But what if some day (heaven forbid) there is some exploit they find? (Unlikely but not impossible).

An ounce of preparation is worth a pound of cure.  I think we need update checking on by default and opt-out only.  We can even put the opt-out checkbox in the above-pictured notification window.

Curious what your thoughts are.

EDIT: See my rationale below: https://github.com/Electron-Cash/Electron-Cash/issues/1142#issuecomment-460433542

**Comment by imaginaryusername** [comment]
The reason I'm against opt-out is: When you download electron-cash, you consent to trusting the repo _now_. But trusting the repo, and hence the code _now_, does not mean you also trust the repo some time in the future; a future evil Calin or Fyookball may use this as a weakness, and even if the user already know the repo is no longer trustworthy they can still be tricked into clicking by the dialog.

Although if there's no one-click install on the dialog I'm quite a bit less opposed to it; up to the others, I guess.

**Comment by cculianu** [comment]
Ah, good reasoning.  Thanks for the feedback.  Note: I won't do this if people are against it.  I'll leave this up in the air and see if anyone else is strongly for or against.

**Comment by markblundeberg** [comment]
I have a different take: I'm fine with opt-out, however, regardless of whether it's opt-out or opt-in, there should be no one-click install. :-P

**Comment by jkister** [comment]
from an end-user's point of view (although i'm on slack and check /r/btc daily) i like the auto-notify of update.  there's a few things we can do to make people think before clicking/accepting the idea - like having them type in "i will be careful" or making them paste in the correct hash of the new release.  we could possibly alleviate future evil colin/jonald concerns by requiring some number of preferred electrumx operators to set some "i agree there's an update" flag.  silly maybe.  just ideas.

**Comment by cculianu** [comment]
Regarding @imaginaryusername 's reservations -- can you please elaborate on what "evil scenario" you have in mind?  

From my perspective it's hard for me to imagine how an update checker does any more harm than what we do now.

Bear in mind that a bad actor would need all **3 of these** in order to use the update checker to publish malicious software:

1. access to a signing key
2. also have to have write access to electroncash.org
3. also have write access to github

If a bad actor takes over (2) and (3) that's still not enough if they lack a signing key.  They need a key too. If anything this update mechanism is about as secure (or even moreso) as the reddit post + electroncash.org trust we rely on now.

Please do elaborate.

@jkister Hmm.  The electumx server coalition idea is very democratic -- but it may be a huge headache... Hmm. It is democratic though -- sort of consensus-based.

**Comment by cculianu** [comment]
Sorry for the spam but I felt I needed to say this:

My rationale for auto-ON (opt-out) is basically that precisely the people that are most vulnerable -- the non-power users -- are precisely the people that will NOT notice the update checker's existence.

I want us to be able to quickly notify non-power users in the future should an exploit arise that needs people to upgrade ASAP.

It is my feeling that having it be auto-on (you have to uncheck a checkbox to opt out) would maximize the probability that a novice user who doesn't go digging into settings would get notified and would benefit maximally from this system.

By leaving it off-by-default -- exactly the most vulnerable people will not notice this setting and will not benefit from it.

Us power users are on reddit and are in touch with the grapevine.  This feature isn't for us.  It's for them.

**Comment by cculianu** [comment]
Ok, I finally got Jonald's ok on making it opt-out.  Perhaps to my credit (or discredit?), Jonald initially assumed this auto-installed the update... hence his not wanting it.

As we chatted it became clear he thought this does more than it actually does -- and he was like the 4th person to think that.  Note to self: So clearly I need to go back and every time I announce something be much less wordy and much clearer in explaining what I intend to do.. 

Anyway -- I'm going to enable it as opt-out.  Closing issue.
